### ISSUE TITLE
import category image has wrong path

### ISSUE THREAD

**Issue by gregordonsky** [techdivision/import#156]
import-cli-simple 3.4.1
magento 2.3.2

I try to import category images in magento path: pub/media/catalog/category
from another folder which have nested subfolders.

a small piece of my json configuration:

```
            {
              "id": "import_category.subject.bunch",
              "identifier": "files",
              "file-resolver": {
                "prefix": "category-import"
              },
              "filesystem-adapter" : {
                "id" : "import.adapter.filesystem.factory.league",
                "adapter" : {
                  "type" : "League\\Flysystem\\Adapter\\Local"
                }
              },
              "params" : [
                {
                  "copy-images" : true,
                  "media-directory" : "pub/media/catalog/category",
                  "images-file-directory" : "process_csv/repository/immagini"
                }
              ],
              "observers": [
                {
                  "import": [
                    "import_category.observer.composite.add_update"
                  ]
                }
              ]
            },

```

example of value in image_path column in CSV:
`stili/classic/classic_bombay.jpg`

Images and folders are successfully copied from
process_csv/repository/immagini
to
pub/media/catalog/category

but in DB every image path record start with a leading / 
ex. `/stili/classic/classic_bombay.jpg`
so for Magento the path is absolute and not relative .

https://github.com/techdivision/import/blob/f481b608b9f76fb96eb857cb727b1654c721dab0/src/Subjects/FileUploadTrait.php#L253-L260

Suggested fix:
- trim all / for a path... just in case there is a trailing slash in configuration and then re-append it.

```
        // trim the leading /, if available
        $trimmedFilename = ltrim($filename, '/');
        $mediaDir = trim($this->getMediaDir(), '/') . '/';
        $imagesFileDir = trim($this->getImagesFileDir(), '/') . '/';

        // prepare source/target filename
        $sourceFilename = sprintf('%s%s', $imagesFileDir, $trimmedFilename);
        $targetFilename = sprintf('%s%s', $mediaDir, $trimmedFilename);
```

**Comment by wagnert** [comment]
@gregordonsky Thanks for reporting. I'll check this as soon as possible and come back to you.

**Comment by gregordonsky** [comment]
Thank you @wagnert .
I re-test my solution and it's wrong.

When i upload image from Magento backend seems that:
- product image path is relative to product media dir in DB records   ( ex. `/8/1/81031100.png` ) with dispersion path active
- swatch option attribute image path is relative to attribute swatch media dir in DB records   (ex. `/8/1/81031100.png` ) with dispersion path active
- category image path is:
  1. relative to Magento dir in DB records  (ex. `/pub/media/catalog/category/stili/classic/classic_bombay.jpg` ) with dispersion path inactive 
  2. is only the filename in DB records (ex. `classic_bombay.jpg` )

So my solution break the import product image functionality

Unfortunatly, i don't know if the different behaviour for categories it is a bug only in my Magento application, or is a general problem.

**Comment by wagnert** [comment]
@gregordonsky Sorry for not answering on this issue. I'll also try to check how Magento handles category images, whereas I thought in our category example files is a working row with images. But I'll also check it and come back to you as soon as possible.

**Comment by wagnert** [comment]
@gregordonsky I've fixed the issue with version 3.7.2. It'll be nice if you would give it a try. I've also added a short description of image import in our documentation under https://docs.m2if.com/getting-started/configuration#images. I hope that helps a bit.
